### DISCUSSION TITLE
Extend SalesChannelLocationService.listLocationIds with more parameters

### DISCUSSION THREAD

**Discussion by khanh-to-niteco** [medusajs/medusa#4766] [Feature Requests]
We would like to extend the function listLocationIds in SalesChannelLocationService with more parameters. Ideally if a context is passed to this method, all we have to do is to extend that context’s class with additional parameters, then overriding listLocationIds () to add new logic, without having to extend any other class or method.

 

**Question**: Can the method confirmInventory in ProductVariantInventoryService and listLocationIds in SalesChannelLocationService be adapted to take an extensible context class as parameter?

```
async confirmInventory(

  variantId: string,

  quantity: number,

   context: { salesChannelId?: string | null} = {}

 ): Promise<Boolean> {

   if (!variantId) {

     return true

   }

 

   ……

 

   let locationIds: string[] = []

   if (context.salesChannelId) {

    locationIds = await this.salesChannelLocationService_.listLocationIds(

       context.salesChannelId

      )
```

**Comment by vholik** [comment]
I think yes.

**Comment by srindom** [comment]
There is no straightforward way to do this as is, but maybe we can find a solution if you can provide a few more details:

- Could you explain the problem that gives rise to this need? 
- What additional parameters would you like to pass, and what kind of logic do you intend to extend the Service with?
